src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs

```diff
@@ -73,6 +74,24 @@ private static bool GetIsWindowsSubsystemForLinux()
             return false;
         }
 
+        static Lazy<bool> s_largeArrayIsNotSupported = new Lazy<bool>(IsLargeArrayIsNotSupported);
+
+        [MethodImpl(MethodImplOptions.NoOptimization)]
+        static bool IsLargeArrayIsNotSupported()
+        {
+            try
+            {
+                var tmp = new byte[int.MaxValue];
+                return tmp == null;
+            }
+            catch (OutOfMemoryException)
+            {
+                return true;
+            }
+        }
+
+        public static bool IsNotIntMaxValueArrayIndexSupported => s_largeArrayIsNotSupported.Value;
```

Comment:
What you have is fine, but does it need to use lazy?  I'm wondering if it can just be:
```C#
public static bool IsNotIntMaxValueArrayIndexSupported { get; } = IsLargeArrayIsNotSupported();
```
or if there's a reason doing that in the static cctor is a bad idea?

Reply:
It could be but allocating 2GB of memory when any of other values are read from the type does not seem to me like a good idea

Reply:
Ah, this is in the PlatformDetection class; in that case, I agree.  I was thinking it was on the test class itself, in which case such a benefit wouldn't exist.